src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -1649,68 +1651,21 @@ protected virtual CultureInfo Culture
             }
         }
 
-        private static void EmitDeterminismKey(CommandLineArguments args, string[] rawArgs, string baseDirectory, CommandLineParser parser)
+        private void EmitDeterminismKey(
+            Compilation compilation,
+            ImmutableArray<AdditionalText> additionalTexts,
+            ImmutableArray<DiagnosticAnalyzer> analyzers,
+            ImmutableArray<ISourceGenerator> generators,
+            ImmutableArray<KeyValuePair<string, string>> pathMap,
+            EmitOptions? emitOptions)
         {
-            var key = CreateDeterminismKey(args, rawArgs, baseDirectory, parser);
-            var filePath = Path.Combine(args.OutputDirectory, args.OutputFileName + ".key");
+            var key = compilation.GetDeterministicKey(additionalTexts, analyzers, generators, pathMap, emitOptions);
```

Comment:
Generators can change the code they produce based on the values passed in through analyzer config files, so I think we need to include those too.

Reply:
This potentially overlaps with some of the work I mentioned here: https://github.com/dotnet/roslyn/pull/57177 

We may consider adding equality for the `AnalyzerConfigOptionsProvider`, in which case we wouldn't need to consider the input analyzer config files themselves, but instead the parsed values according to the rules of editorconfig, which would allow for functionally equivalent edits to still be treated as equal.